src/libaktualizr/primary/reportqueue_test.cc

```diff
@@ -122,6 +136,32 @@ TEST(ReportQueue, FailureRecovery) {
   EXPECT_EQ(http->events_seen, num_events);
 }
 
+/* Test event store. */
+TEST(ReportQueue, StoreEvents) {
+  TemporaryDirectory temp_dir;
+  Config config;
+  config.storage.path = temp_dir.Path();
+  config.tls.server = "";
+
+  auto sql_storage = std::make_shared<SQLStorage>(config.storage, false);
+  size_t num_events = 10;
+  {
+    auto http1 = std::make_shared<HttpFakeRq>(temp_dir.Path(), num_events);
+    ReportQueue report_queue1(config, http1, sql_storage);
+    for (size_t i = 0; i < num_events; ++i) {
+      report_queue1.enqueue(std_::make_unique<EcuDownloadCompletedReport>(
+          Uptane::EcuSerial("StoreEvents" + std::to_string(i)), "", true));
+    }
+  }
+
+  config.tls.server = "reportqueue/StoreEvents";
+  auto http = std::make_shared<HttpFakeRq>(temp_dir.Path(), num_events);
+  ReportQueue report_queue(config, http, sql_storage);
+  // Wait at most 30 seconds for the messages to get processed.
+  http->expected_events_received.get_future().wait_for(std::chrono::seconds(20));
+  EXPECT_EQ(http->events_seen, num_events);
```

Comment:
This looks really good! My only concern is that we should check that there is actually something in the database before restarting the `ReportQueue`. We aren't testing anything interesting if it turns out that the first `ReportQueue` was actually able to send out all of the events before it got shut down. Basically I just want to make sure the test is testing what we think it is.

Reply:
Ah, because the server url which is used for the first ReportQueue is empty, it could not send any events out. But it's a good idea to check the events number in sql, I added it to the case, and I also checked whether the number was set to 0 after I send them out.